src/Sentry/Laravel/Tracing/Middleware.php

```diff
@@ -103,12 +104,14 @@ private function startTransaction(Request $request, HubInterface $sentry): void
     {
         $requestStartTime = $request->server('REQUEST_TIME_FLOAT', microtime(true));
         $sentryTraceHeader = $request->header('sentry-trace');
+        $baggageHeader = $request->header('baggage');
 
         $context = $sentryTraceHeader
-            ? TransactionContext::fromSentryTrace($sentryTraceHeader)
+            ? TransactionContext::fromHeaders($sentryTraceHeader, $baggageHeader)
             : new TransactionContext;
 
         $context->setOp('http.server');
+        $context->setSource(TransactionSource::url());
```

Comment:
@stayallive Could we also set this to `route`?
Are there cases where the transaction name could be `/users/edit/123` vs `/users/edit/:user_id`?

Reply:
So it turns out, yes, yes it can. `Integration::extractNameAndSourceForRoute` (and it's Lumen variant) will do that (unless a route name was defined or controller class).

But we do that as late as possible so I've removed the `setSource` at that location since we are never actually setting the transaction name.